source/config/profileUpgradeSteps.py

```diff
@@ -134,3 +139,160 @@ def upgradeConfigFrom_7_to_8(profile: Dict[str, str]) -> None:
 			profile['documentFormatting']['reportTableHeaders'] = ReportTableHeaders.ROWS_AND_COLUMNS.value
 		else:
 			profile['documentFormatting']['reportTableHeaders'] = ReportTableHeaders.OFF.value
+
+
+def upgradeConfigFrom_8_to_9(profile: Dict[str, str]) -> None:
+	"""
+	In NVDA config, when various values were controlling a single combobox in the settings window, these values
+	have been replaced by a single value.
+	The following settings are upgraded:
+	- Line indentation (in Document formatting settings)
+	- Show messages (in Braille settings)
+	"""
+	
+	_upgradeConfigFrom_8_to_9_lineIndent(profile)
+	_upgradeConfigFrom_8_to_9_cellBorders(profile)
+	_upgradeConfigFrom_8_to_9_showMessages(profile)
+	_upgradeConfigFrom_8_to_9_tetherTo(profile)
+
+
+def _upgradeConfigFrom_8_to_9_lineIndent(profile: Dict[str, str]) -> None:
+	anySettingInConfig = False
+	try:
+		reportLineIndent: str = profile['documentFormatting']['reportLineIndentation']
+		anySettingInConfig = True
+	except KeyError:
+		reportLineIndent = False
+	try:
+		reportLineIndentWithTones: str = profile['documentFormatting']['reportLineIndentationWithTones']
+		del profile['documentFormatting']['reportLineIndentationWithTones']

```

Comment:
I agree that keeping the old keys in the config does not cause any problem.
But I prefer deleting the old keys to clarify things for developers/testers when reading the config.

Regarding backward compatibility, even if I do not remove the old keys, it is not guaranteed by this PR as written today. For example `profile['documentFormatting']['reportLineIndentation']` can be upgraded to a value of 3 (speech and tones) with schema version = 9 whereas it can only be True or False with schema version = 8. I would have to avoid reusing old key names to avoid this issue.

Moreover, backward compatibility of configuration has never been a requirement and it has always been discouraged to downgrade NVDA. If there is an evolution on this topic, it should be acted by NVAccess. @seanbudd could you clarify NVAccess' position?

Reply:
>Moreover, backward compatibility of configuration has never been a requirement and it has always been discouraged to downgrade NVDA. If there is an evolution on this topic, it should be acted by NVAccess. @seanbudd could you clarify NVAccess' position?

This is still true. I think deleting the keys is probably the safer option for the long term.

Reply:
Config is backed up before being upgraded. The safest way to "downgrade" would be to restore one of those config-files.
However, there are many UX edge cases to consider. I don't think the value for users is worth the extra development complexity.

Reply:
OK. So I do not modify anything.